nextflow/modules/CONVERT/templates/convert.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -o pipefail
+set -eu
+
+    if [[ "!{meta.fileType1}" == "OPENARRAY" ]]
+    then
+        oafile="!{files[0]}"
+        sampleId="!{meta.data1Id}"
+        if [[ "!{files[1]}" == *".gz" ]]
+        then
+            cp "!{files[1]}" "!{vcf2}"
```

Comment:
hoe werkt dit precies, want vcf1 en vcf2 zijn volgens convert.nf `vcf1="${meta.data1Id}.converted.vcf.gz"` en `vcf2="${meta.data2Id}.converted.vcf.gz"`
files[1] wordt gekopieerd of indien het nog geen .gz file is, word ie ge-bgzipped.
Helemaal onderin het script schrijf je je output weg naar `${sampleId}.converted.vcf`, die je daarna ook bgzipped.
Dus eerst kopieer/bgzip je de file om um daarna te laten overschrijven door het `array-as-vcf` script?

Reply:
Ja dit stukje verdient niet de schoonheidprijs, en ik moet hier nog een betere oplossing voor vinden, maar dat is nog niet gelukt zonder hele pipeline te moeten verbouwen. Maar het probleem wat ik hier tegenaanloop is dat er 2 inputfiles in dit script gaan, waarbij je met 1 maar wat mee doet. alleen de openarray file. Met de vcf gebeurd niks. Alleen weet ik niet of file 1 een vcf is of file 2. Maar ik mag niet niks doen met file 2 de vcf, vandaar dat ik die kopieer. Anders stuitert nf. En ik moet ook checken of het wel een .vcf of .vcf.gz is, anders wordt de file als .gz weggeschreven zonder dat die ingepakt is. Dan krijg je verderop weer problemen dat het een niet ingepakt file is met een .gz extentie. Het werk voor nu, maar het moet nog een keer anders denk ik.